### ISSUE TITLE
[REVIEW]: dantro: a Python package for handling, transforming, and visualizing hierarchically structured data

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2316]
**Submitting author:** @blsqr (<a href="http://orcid.org/0000-0003-3858-0904">Yunus Sevinchan</a>)
**Repository:** <a href="https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro" target ="_blank">https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro</a>
**Version:** v0.13
**Editor:** @lorenanicole
**Reviewer:** @Chilipp, @mdpiper
**Archive:** Pending

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/26df6e37a27fc211889660381772dac8/status.svg)](https://joss.theoj.org/papers/26df6e37a27fc211889660381772dac8)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/26df6e37a27fc211889660381772dac8"><img src="https://joss.theoj.org/papers/26df6e37a27fc211889660381772dac8/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/26df6e37a27fc211889660381772dac8/status.svg)](https://joss.theoj.org/papers/26df6e37a27fc211889660381772dac8)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@Chilipp & @mdpiper, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @lorenanicole know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@blsqr) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@blsqr) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @Chilipp, @mdpiper it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2316 with the following error: 

 Can't find any papers to compile :-(

**Comment by lorenanicole** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2316 with the following error: 

 Can't find any papers to compile :-(

**Comment by lorenanicole** [comment]
Hmmm @arfon could you 👀 why the pdf is failing to compile?

**Comment by blsqr** [comment]
That's on us. We have the paper on a separate branch, `add-joss-paper`, such that we can add changes there and merge that branch once feedback was added.

**Comment by blsqr** [comment]
@whedon generate pdf from branch add-joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch add-joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02316/joss.02316/10.21105.joss.02316.pdf)

**Comment by lorenanicole** [comment]
Thanks @blsqr for that!

**Comment by lorenanicole** [comment]
👋  @Chilipp, @mdpiper! You are ready to start review on this issue. Please update the checklist above and provide all feedback in this issue. Also you can find the reviewer guidelines are available here 👉 https://joss.readthedocs.io/en/latest/reviewer_guidelines.html.

**Comment by mdpiper** [comment]
Opened https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/issues/175 re a minor issue in references.

**Comment by mdpiper** [comment]
Opened https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/issues/176 with minor edits to paper text.

**Comment by blsqr** [comment]
Thanks, @mdpiper!

I forgot to mention that we are working on a self-hosted GitLab instance and by default accounts are marked as "external", meaning that freshly-registered users can create issues but can _not_ create their own projects and forks. I will grant you access to the dantro repository so that you have more freedom for future issues or MRs. (@Chilipp, if you could drop me a note once you registered, I'll do the same for your account.)

**Comment by lorenanicole** [comment]
@Chilipp just checking in to confirm if you need anything at the moment!

**Comment by blsqr** [comment]
@whedon generate pdf from branch add-joss-paper

Re-generating the article proof to address the reference issue ([#175](https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/issues/175)) and include the improvements to `paper.md` @mdpiper suggested in [#176](https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/issues/176). Thanks for both!

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch add-joss-paper
. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2316 with the following error: 

 error: pathspec 'add-joss-paper
' did not match any file(s) known to git.
Can't find any papers to compile :-(

**Comment by blsqr** [comment]
@whedon generate pdf from branch add-joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch add-joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02316/joss.02316/10.21105.joss.02316.pdf)

**Comment by Chilipp** [comment]
@lorenanicole, it's all fine. Apologies for the delay. A few urgent things came in today and will take most of the week. So I'll probably won't be able to do the review before Monday next week.

**Comment by lorenanicole** [comment]
Totally understandable! Thank you for the update 🙇‍♀️ , I'll circle back in about a week.

**Comment by mdpiper** [comment]
Both issues I posted upthread have been addressed by @blsqr.

**Comment by mdpiper** [comment]
Hi @lorenanicole, I've finished my review. I'd like to recommend the paper for publication.

@blsqr I enjoying learning about `dantro`. It's a tool I could see using in my work. I especially appreciate that it works out of the box with netCDF files that use the netCDF-4 data model (built on HDF5).

**Comment by lorenanicole** [comment]
Thanks @mdpiper! 

So @blsqr I'll circle back in a week or so to check in on @Chilipp's progress.

**Comment by Chilipp** [comment]
Dear @lorenanicole and @blsqr, I strongly apologize for the long delay. I am done with my review now and `dantro` is obviously in a very nice condition, great job! I opened a couple of small issues in the Gitlab repository and it would be nice if we could address them to help new users getting started with `dantro`. In summary:

### General checks

 > - [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

The repo contains two license files, and I think one of them can be deleted (see [!183](https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/issues/183))(but this is not a blocking issue, so I checked it off)


> - [ ] **Installation:** Does installation proceed as outlined in the documentation?

I strongly recommend to use conda rather than `pip` for the installation instructions (see [!184](https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/issues/184)), as `pip` often causes troubles, e.g. on Windows

### Functionality

> - [ ] **Functionality:** Have the functional claims of the software been confirmed?

I'd like to check this off when worked myself through the *Usage Example* (see below).

### Documentation

> - [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.

Their is no list of dependencies in the docs, only in the `setup.py` script (see [!185](https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/issues/185)).

> - [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).

You do provide a nice Usage Example, but to me it seems like you do not provide the demo files that have been used to generate it. See [!186](https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/issues/185)

**Comment by Chilipp** [comment]
@lorenanicole: all open points from https://github.com/openjournals/joss-reviews/issues/2316#issuecomment-653251333 have been addressed by @blsqr so I also recommend this paper for publication! 

Thanks a lot @blsqr, @jeremiastraub and @herdeanu for your great work here and particularly for your efforts generating a high quality documentation!

**Comment by lorenanicole** [comment]
Excellent! CC: @arfon

**Comment by arfon** [comment]
:tada: great! @lorenanicole, if all the reviewers have recommended the paper and you're ready to move this forward then these are the next steps: https://joss.readthedocs.io/en/latest/editing.html#after-reviewers-recommend-acceptance

**Comment by blsqr** [comment]
Thank you very much for reviewing dantro, @mdpiper and @Chilipp, and for providing highly valuable feedback on how to improve documentation and usage of it! We very much appreciate it!

@lorenanicole We are currently putting the final touches on the `paper.md` (see [MR !163](https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/merge_requests/163)) (updating and cleaning-up of references, cross-links to the publication's DOI, …) and will inform you once we are done with that part. Is there anything else we can do at this point?

A more long-term question: dantro is currently hosted on a GitLab instance at our institute. Some time in 2021, the dantro project will have to move to another host, presumably GitLab.com or GitHub. Will it be possible to update the "Software Repository" links that appear in the PDF and on the JOSS homepage to those new URLs? (Of course we will aim to preserve all links via appropriate redirects, but it is currently unclear in how far that will be possible.)

**Comment by blsqr** [comment]
@lorenanicole We added some minor changes to the manuscript and the documentation, updating references, fixing some small grammar issues, and adding another link regarding the `conda`-based installation of dantro. For details, see [the corresponding commits](https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/merge_requests/163/commits).

I will let `@whedon` create another preview of the paper. If I understand the steps cited above correctly, you would also be going over the manuscript again? In that case, I would wait before merging that MR, such that we can incorporate any further changes on the `add-joss-paper` branch. Is that ok?

After the paper is merged – and if there are no further points that we should address here –, we would be ready to create a tagged release and upload it to zenodo.

**Comment by blsqr** [comment]
@whedon generate pdf from branch add-joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch add-joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02316/joss.02316/10.21105.joss.02316.pdf)

**Comment by blsqr** [comment]
@whedon check references

**Comment by blsqr** [comment]
@lorenanicole @arfon I can't seem to ask whedon to check the references on the current manuscript. Could you perhaps trigger a reference check? Is there anything else we can do at this point?

**Comment by arfon** [comment]
@whedon check references from branch add-joss-paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch add-joss-paper
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/978-3-030-50436-6_32 is OK
- 10.5281/zenodo.3826470 is OK
- 10.5334/jors.148 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.21105/joss.01057 is OK

MISSING DOIs

- https://doi.org/10.25080/majora-7b98e3ed-013 may be missing for title: Dask: Parallel Computation with Blocked algorithms and Task Scheduling
- https://doi.org/10.31219/osf.io/mqzyx may be missing for title: Vega-Lite: A Grammar of Interactive Graphics

INVALID DOIs

- 10.21105/joss.02165 is INVALID
```

**Comment by blsqr** [comment]
Thanks, @arfon! I adjusted the references, see [the MR](https://ts-gitlab.iup.uni-heidelberg.de/utopia/dantro/-/merge_requests/163).

> https://doi.org/10.25080/majora-7b98e3ed-013 may be missing for title: Dask: Parallel Computation …

Added this DOI.

> https://doi.org/10.31219/osf.io/mqzyx may be missing for title: Vega-Lite: A Grammar of Interactive Graphics

The suggested DOI seems to be that of the OSF preprint. I found the actual DOI (https://doi.org/10.1109/TVCG.2016.2599030) and added it to the references

> 10.21105/joss.02165 is INVALID

This is a citation of a work where dantro is prominently used in, called *Utopia*, which is currently also [under review for JOSS](https://github.com/openjournals/joss-reviews/issues/2165).  
We would very much like to cite the Utopia framework, and – if accepted – its JOSS publication seems most suited for that purpose.

How should this case be handled? Should we adapt the citation to not include the DOI or choose a different reference altogether? Or can we wait until the review process for Utopia is finished?

**Comment by arfon** [comment]
> This is a citation of a work where dantro is prominently used in, called Utopia, which is currently also under review for JOSS.
We would very much like to cite the Utopia framework, and – if accepted – its JOSS publication seems most suited for that purpose.

We can cite it here (as you currently have it). Whedon is just flagging this DOI as not (yet) valid.

**Comment by blsqr** [comment]
Glad to hear, thanks 👍 

@arfon We were also wondering about another publication-related question (see [above](https://github.com/openjournals/joss-reviews/issues/2316#issuecomment-656691584)):

> A more long-term question: dantro is currently hosted on a GitLab instance at our institute. Some time in 2021, the dantro project will have to move to another host, presumably GitLab.com or GitHub. Will it be possible to update the "Software Repository" links that appear in the PDF and on the JOSS homepage to those new URLs? (Of course we will aim to preserve all links via appropriate redirects, but it is currently unclear in how far that will be possible.)

**Comment by arfon** [comment]
> A more long-term question: dantro is currently hosted on a GitLab instance at our institute. Some time in 2021, the dantro project will have to move to another host, presumably GitLab.com or GitHub. Will it be possible to update the "Software Repository" links that appear in the PDF and on the JOSS homepage to those new URLs? (Of course we will aim to preserve all links via appropriate redirects, but it is currently unclear in how far that will be possible.)

We have done this before and so do our best to help with this.

**Comment by blsqr** [comment]
@lorenanicole We were wondering how we should proceed here? Could you perhaps give us a brief update? The current status is summarised in [the comment above](https://github.com/openjournals/joss-reviews/issues/2316#issuecomment-658135716).

**Comment by blsqr** [comment]
@arfon We can't seem to reach @lorenanicole via this channel. As far as I see, we would need to get her feedback on the manuscript before we can proceed to make a release and upload a snapshot of dantro to zenodo.

**Comment by arfon** [comment]
👍 thanks for the ping @blsqr - I just emailed @lorenanicole to ask her to check in soon.

**Comment by lorenanicole** [comment]
Hi hi hi @blsqr thanks for reaching out! Checking in on the status now, thanks @arfon for confirming the DOI status.

**Comment by lorenanicole** [comment]
@blsqr tonight I'll review the paper again and update when I have done so. Provided, minus the DOI noted issues, all is well there I'll move towards accepting for final review.

**Comment by blsqr** [comment]
@lorenanicole Perfect, thanks for the update! 😊

**Comment by lorenanicole** [comment]
Of course!

**Comment by lorenanicole** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2316 with the following error: 

 Can't find any papers to compile :-(

**Comment by lorenanicole** [comment]
@whedon generate pdf from branch add-joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch add-joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02316/joss.02316/10.21105.joss.02316.pdf)

**Comment by lorenanicole** [comment]
@blsqr did another re-read and we look good ✅ minus the noted DOI on the  Utopia. please archive a tagged release of your submission on an archive site such as Zenodo or figshare, and post the DOI for the archive to this issue thread. Finally, ensure that the version number for the submission at the top of this issue matches the version of the archived code. If it does not, let me know what the correct version number should be so I can set it.

**Comment by blsqr** [comment]
Glad to hear! Thanks, @lorenanicole! :)

We created a tagged release, v0.14.0, and uploaded it [to zenodo](https://zenodo.org/record/3986628). The corresponding DOI is: `10.5281/zenodo.3986628`

> Validate the version number for the submission at the top of this issue matches the version of the archived code.

The version number indeed _changed_ to `v0.14.0`.

**Comment by lorenanicole** [comment]
@whedon set 10.5281/zenodo.3986628 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3986628" target="_blank">10.5281/zenodo.3986628</a> is the archive.

**Comment by lorenanicole** [comment]
@whedon set v0.14.0 as version

**Comment by whedon** [comment]
OK. v0.14.0 is the version.

**Comment by lorenanicole** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/978-3-030-50436-6_32 is OK
- 10.5281/zenodo.3826470 is OK
- 10.5334/jors.148 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.25080/majora-7b98e3ed-013 is OK
- 10.21105/joss.01057 is OK
- 10.1109/TVCG.2016.2599030 is OK

MISSING DOIs

- None

INVALID DOIs

- 10.21105/joss.02165 is INVALID
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1647

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1647, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by lorenanicole** [comment]
@blsqr feel free to review https://github.com/openjournals/joss-reviews/issues/2316#issuecomment-675062999 as well!

**Comment by blsqr** [comment]
@lorenanicole Looks good to us! 😊

**Comment by lorenanicole** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @lorenanicole, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by lorenanicole** [comment]
Oops. Heh! My bad, we're just waiting on the EiCs now @blsqr!

**Comment by blsqr** [comment]
Big thanks from us, too! We really enjoyed the review process and are grateful for the feedback we got on dantro. 😊

**Comment by arfon** [comment]
> Re-rendering would not only apply to this paper but also to the ones reviewed in https://github.com/openjournals/joss-reviews/issues/2165 and https://github.com/openjournals/joss-reviews/issues/2313 which were hosted on that same server. Should we create separate posts in the respective issues for that?

Is the exact same paper present in the repository (https://gitlab.com/utopia-project/dantro) as was the case when the original paper was published? Also, for the other two papers?

**Comment by blsqr** [comment]
Yes, the papers are exactly the same for both dantro ([history](https://gitlab.com/utopia-project/dantro/-/commits/master/joss) only contains a deleted whitespace) and Utopia ([history](https://gitlab.com/utopia-project/utopia/-/commits/master/joss)).

The DORiE paper has no changes either ([history](https://ts-gitlab.iup.uni-heidelberg.de/dorie/dorie/-/commits/master/joss)) but the repository has not yet been fully migrated; @peanutfun is it clear when this would be concluded?

**Comment by peanutfun** [comment]
Migration of DORiE will probably conclude within this week.

@blsqr Thanks a lot for bringing this up! 🙌 

@soilros fyi, with DORiE and our other repos migrating from our GitLab instance we would like to update the JOSS papers with the new repository links

**Comment by peanutfun** [comment]
DORiE was migrated to https://gitlab.dune-project.org/dorie/dorie

**Comment by blsqr** [comment]
@arfon With that, we have all the projects migrated:

* **dantro**: https://gitlab.com/utopia-project/dantro
* **Utopia**: https://gitlab.com/utopia-project/utopia
* **DORiE**: https://gitlab.dune-project.org/dorie/dorie

They all have their (unchanged) papers located at `joss/paper.md`. Is there anything else we need to do?

**Comment by blsqr** [comment]
@arfon Just pushing this again, because our servers will soon have to shut down. How would we go about re-rendering the papers and the project pages?

**Comment by arfon** [comment]
@editorialbot set https://gitlab.com/utopia-project/dantro as repository

**Comment by arfon** [comment]
@editorialbot reaccept

**Comment by arfon** [comment]
@xuanxu – could you assist here? I'm trying to republish a paper from a couple of years ago with a new software URL but the updated paper has a few issues:

- The published date on the site now lists this paper as being published today (31st July 2022) rather than August 2020).
- The citation string generated by the reaccept command (on the JOSS site) appears to also think the paper was published today.
- The issue/volume information is wrong `7(75)` on the paper.

Essentially it looks like the reaccept command isn't able to handle the volume/issue/citation string information correctly for older papers. Is this something that is meant to work properly?

**Comment by blsqr** [comment]
Thanks for looking into this, @arfon and @xuanxu!

Once the dantro publication date has been fixed and everything works as expected: Should we add separate posts in the review issues of the other two affected publications?

**Comment by blsqr** [comment]
@arfon Sorry for pushing this again :/ Would you mind updating the JOSS database entry and then re-running the `reaccept` command on this paper? Otherwise the paper information remains wrong.

**Comment by blsqr** [comment]
@arfon @xuanxu @labarba Any chance you could give this another look? The citation for our paper has now been wrong for three weeks.

I don't suppose there's anything that we could help with, right?

**Comment by arfon** [comment]
@editorialbot reaccept

**Comment by arfon** [comment]
@editorialbot reaccept

**Comment by arfon** [comment]
@editorialbot reaccept

**Comment by arfon** [comment]
@blsqr – looks like this paper is fixed up now. Can you email me directly the other papers that need reprocessing? arfon.smith@gmail.com

**Comment by blsqr** [comment]
Thank you so much, @arfon! Looks great 🎉 

> Can you email me directly the other papers that need reprocessing?

Will do 👍
